src/main/java/org/apache/jackrabbit/filevault/maven/packaging/GenerateMetadataMojo.java

```diff
@@ -1019,15 +1029,46 @@ String getModuleArtifactKey(final Artifact artifact) {
                 getLog().info(String.format("Embedding %s (from %s) -> %s", artifact.getId(), source.getAbsolutePath(), targetPathName));
                 fileMap.put(targetPathName, source);
                 if (pack.isFilter()) {
-                    addWorkspaceFilter(targetNodePathName);
+                    addEmbeddedFileToFilter(targetNodePathName, pack.isAllVersionsFilter());
                 }
             }
         }
         return fileMap;
     }
 
-    private void addWorkspaceFilter(final String filterRoot) {
-        filters.add(new PathFilterSet(filterRoot));
+    private void addEmbeddedFileToFilter(String embeddedFile, boolean includeAllVersions) throws ConfigurationException {
+        filters.add(getPathFilterSetForEmbeddedFile(embeddedFile, includeAllVersions));
+    }
+
+    static PathFilterSet getPathFilterSetForEmbeddedFile(final String embeddedFile, boolean includeAllVersions)
+            throws ConfigurationException {
+        final PathFilterSet pathFilterSet;
+        if (includeAllVersions) {
+            String filename = FilenameUtils.getName(embeddedFile);
+            // shorten the filter root by one level
+            String rootName = StringUtils.chomp(embeddedFile, "/");
+            String extension = FilenameUtils.getExtension(filename);
+
+            // now find a mattern which should apply to embeddedFile and all similar artifacts with different versions from format:
+            // ${artifactId}-${version})
+            Matcher matcher = FILENAME_PATTERN_WITHOUT_VERSION_IN_GROUP1.matcher(filename);
+            if (!matcher.matches()) {
+                throw new IllegalArgumentException("Could not figure out version part in filename '" + filename
+                        + "'. For this artifact you cannot use 'isAllVersionsFilter=true'");
+            }
+
+            // create new pattern which matches the same artifacts in all versions
+            String pattern = Pattern.quote(matcher.group(1)) + ".*" + "\\." + extension;
+            if (!filename.matches(pattern)) {
+                throw new IllegalArgumentException("Detected pattern '" + pattern + "' does not even match given filename '" + filename
+                        + "'. For this artifact you cannot use 'isAllVersionsFilter=true'");
+            }
+            pathFilterSet = new PathFilterSet(rootName);
+            pathFilterSet.addInclude(new DefaultPathFilter(pattern));
```

Comment:
I think you need to add the _match-node-and-all-children_ pattern, just to be sure:

```
pattern += '(/.*)?`
```

Reply:
Good point, gonna fix this.